client/state/data-layer/wpcom/sites/media/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Internal dependencies
+ */
+import { MEDIA_REQUEST } from 'state/action-types';
+import { isRequestingMedia } from 'state/selectors';
+import { failMediaRequest, receiveMedia, requestingMedia } from 'state/media/actions';
+import wpcom from 'lib/wp';
+import debug from 'debug';
+
+/**
+ * Module variables
+ */
+const log = debug( 'calypso:middleware-media' );
+
+/**
+ * Issues an API request to fetch media for a site and query.
+ *
+ * @param  {Object}  store  Redux store
+ * @param  {Object}  action Action object
+ * @return {Promise}        Promise
```

Comment:
While I appreciate the comment `Promise` is somewhat redundant :wink: Maybe something a bit more descriptive could help here, or we could just leave it off.

```js
/**
 * @return {Promise} wpcom.js request response
 */
```